skycatalogs/utils/star_parquet_input.py

```diff
@@ -1,33 +1,47 @@
-# import pyarrow as pa
 import os
+import glob
 import pyarrow.parquet as pq
 import pandas as pd
 import numpy as np
 import numpy.ma as ma
 import healpy
+from lsst.meas.algorithms.htmIndexer import HtmIndexer
 
 
-def _find_star_files(dirpath, pixel):
-    '''
-    Parameters
-    ----------
-    dirpath    string       Path to directory containing input parquet files
-                            HTM partitioning
-    pixel      int          Healpix pixel
+class UWStarFiles:
+    _files = {}  # shared state
 
-    Returns
-    -------
-    List of parquet file paths for files which might have sources belonging
-    to specified pixel
-    '''
-    # For now, hard-code to 3 files which have everything needed
-    # for joint simulation region
+    def __init__(self, input_dir):
+        self._index_files(input_dir)
+        self.htm_indexer = HtmIndexer(depth=20)
 
-    fnames = ['stars_chunk_8800000000000_8900000000000.parquet',
-              'stars_chunk_9200000000000_9300000000000.parquet',
-              'stars_chunk_9700000000000_9800000000000.parquet']
+    def _index_files(self, input_dir):
+        if self._files:
+            # The files have already been indexed.
+            return
+        files = sorted(glob.glob(os.path.join(input_dir,
+                                              'stars_chunk_*.parquet')))
+        for item in files:
+            tokens = os.path.basename(item).split('_')
+            imin = int(tokens[2])
+            imax = int(tokens[3].split('.')[0])
+            self._files[(imin, imax)] = item
 
-    return [os.path.join(dirpath, fname) for fname in fnames]
+    def find_files(self, pixel, nside=32, res_factor=16):
```

Comment:
A docstring for this routine would be welcome.  Why was 16 chosen as default for `res_factor`?

Reply:
What does resfactor have to do with HTM depth (apparently 20), if anything? 
Healpixels with nside=32 are much bigger than HTM depth 20 regions (or depth 16 regions, for that matter). How did you verify that nside*16 gives a sufficiently fine grid?

Reply:
Another way to do this (which seems safer to me) would be to define a minimum circle around the healpixel, then call HtmIndexer.getShardIds